### PR TITLE
[Xamarin.Android.Build.Tasks] Use Microsoft.Android.Build.BaseTasks

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://github.com/xamarin/xamarin-android-tools/pull/101
Context: https://github.com/xamarin/monodroid/pull/1145

We would like to eventually remove the dependency that xamarin/monodroid
has on xamarin/xamarin-android.  As a first step towards this effort,
shared MSBuild code is being moved to xamarin/xamarin-android-tools.

`Xamarin.Android.Build.Tasks` has been updated to use the new shared
`Microsoft.Android.Build.BaseTasks` sources.  All of the files that were
moved to xamarin/xamarin-android-tools have been deleted.

The inverted monodroid build has been replaced with the new monodroid
build system, see https://github.com/xamarin/monodroid/pull/1145 for
more context.  The one build task test that was still being ran from the
xamarin/monodroid repo has been moved into `DebuggingTasksTests.cs`,
which is conditionally included in `Xamarin.Android.Build.Tests`.

Tests associated with the build task source that was moved to
xamarin/xamarin-android-tools have also been moved to a new 
`Microsoft.Android.Build.BaseTasks-Tests` assembly in that repo.

### CODE REVIEW

File: build-tools/installers/create-installers.targets [hunk 1]
```diff
@@ -410,6 +414,7 @@
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Analysis.Tasks.pdb" />
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Android.Build.Debugging.Tasks.dll" />
     <_MSBuildFiles Include="$(MSBuildSrcDir)\Xamarin.Android.Build.Debugging.Tasks.pdb" />
+    <_MSBuildFiles Include="@(_LocalizationLanguages->'$(MSBuildSrcDir)\%(Identity)\Xamarin.Android.Build.Debugging.Tasks.resources.dll')" />
```
Comment (jonathanpeppers) [inline build-tools/installers/create-installers.targets:417]: Whoops, it seems like we might want to fix this one in another PR and `Xamarin.Android.Tools.AndroidSdk.resources.dll` if they need to be in 16.9?

**Review by jonathanpeppers — COMMENTED** [review]
Otherwise, I looked through the diff. Looks good so far. 👍

File: build-tools/installers/create-installers.targets (hunk 1 above)
Comment (pjcollins) [reply build-tools/installers/create-installers.targets:417]: Thanks, I moved these into https://github.com/xamarin/xamarin-android/pull/5481.

File: src/Xamarin.Android.Build.Tasks/Tasks/ConvertDebuggingFiles.cs [hunk 2]
```diff
@@ -28,7 +29,7 @@ public override bool RunTask ()
 					continue;
 
 				try {
-					MonoAndroidHelper.SetWriteable (pdb);
+					Microsoft.Android.Build.Tasks.Files.SetWriteable (pdb);
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Tasks/ConvertDebuggingFiles.cs:32]: does this need to be fully qualified?

Comment (pjcollins) [reply src/Xamarin.Android.Build.Tasks/Tasks/ConvertDebuggingFiles.cs:32]: This property was interfering: https://github.com/xamarin/xamarin-android/blob/060be4b6fdc040ce7a9af238d5b28af624cea46e/src/Xamarin.Android.Build.Tasks/Tasks/ConvertDebuggingFiles.cs#L17

**Review by dellis1972 — APPROVED** [review]

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**